test/export/test_export.py

```diff
@@ -2528,6 +2529,41 @@ def forward(self, x):
         # this doesn't work today
         gm_unflat_strict = unflatten(ep)
 
+@unittest.skipIf(not torchdynamo.is_dynamo_supported(), "dynamo isn't support")
+class TestOneOffModelExportResult(TestCase):
+
+    def test_scaled_dot_product_attention(self):
+        """
+        This test makes sure we are always getting the same decomposition result for SDPA.
+        As of now _scaled_dot_product_flash_attention_for_cpu is expected to show up in
+        export() result. Some downstream backend then further decompose it into core ATen
+        ops in torch/_decomp/decompositions.py (search for
+        _scaled_dot_product_flash_attention_for_cpu).
+
+        Export is decomposing based on the CompositeImplicitAutograd kernel implementation
+        of SDPA. If this test fails, it means the kernel is being modified. In this case
+        we strongly encourage you to change the decomposition rule under
+        torch/_decomp/decompositions.py along with the kernel changes, so all of the
+        downstream backends are not being affected.
+        """
+        class ScaledDotProductAttention(torch.nn.Module):
```

Comment:
Can we also add a test for "cuda" flow, or does that not matter for export?

For instance I imagine that: https://github.com/pytorch/pytorch/pull/115663, would break export of sdpa for gpu.
And that the fix would look like: https://github.com/pytorch/pytorch/pull/117097

I would hope that this would error on CI/CD and then we can fix in the PR

Reply:
Added